analyses/country_template/docs/trigger_report_template.Rmd

```diff
@@ -0,0 +1,333 @@
+hist_period <- "XXXX - XXXX" # eg: 1980 - 2021
+perf_metrics_filename <- "dummy_perf_metrics_table_for_template.csv" # include path if not sister to Rmd
+learning_table_filename <- "dummy_learning_table_for_template.csv" # include path if not sister to Rmd
+
+# trigger A info  
+type_a <- 'eg: Predictive' 
+timepoints_a<- 'eg: Jan, Feb, Mar'
+package_a <- 'TBC' # input number only, no $ sign or 'm' for million
+funding_a <- 'X'
+predicted_window_a <- 'eg: JAS' 
+lead_time_a <- 'eg: 7-5 months' # change units as needed. Include all lead times if multiple activation timepoints
+
+# trigger B info  
+type_b <- 'eg: Predictive'
+timepoints_b <- 'eg: Apr, May, Jun'
+package_b <- 'TBC'
+funding_b <- 'X' 
+predicted_window_b <- 'eg: ASO'
+lead_time_b <- 'eg: 4-1 months'
+
+# trigger C info  
+type_c <- 'eg: Observational'
+timepoints_c <- 'eg: Jul'
+package_c <- 'TBC'
+funding_c <- 'X'
+predicted_window_c <- 'N/A'
+lead_time_c <- 'N/A'
+
+```
+
+```{r setup, include = FALSE, message = FALSE, warning = FALSE}
+knitr::opts_chunk$set(echo = FALSE) # do not print code by default
+knitr::opts_chunk$set(include = TRUE) # include chunk output by default
+knitr::opts_chunk$set(message = FALSE) # do not print messages by default
+knitr::opts_chunk$set(warning = FALSE) # do not print warnings by default
+
+options(scipen = 999) # turn off scientific notation
+options(encoding = "UTF-8") # set encoding to UTF-8 instead of ANSI
+```
+
+```{r libraries, include = FALSE}
+packages <- c('tidyverse', 'ggthemes', 'kableExtra', 'knitr', 'flextable')
+
+# install packages not yet installed
+installed_packages <- packages %in% rownames(installed.packages())
+
+if (any(installed_packages == FALSE)) {
+  install.packages(packages[!installed_packages])
+}
+
+# load libraries 
+lapply(packages, library, character.only = TRUE)
+
+```
+
+```{r functions}
+computeCIwidths <- function(metric_name, trigger_id) { 
+
+  perf_metrics_data %>%
+  filter(trigger == trigger_id) %>%
+  select(-trigger) %>%
+  filter(metric == metric_name) %>%
+  pivot_wider(names_from = point, values_from = value) %>%
+  mutate(below_ci = low_end,  # width between zero and low end of CI
+         ci = high_end - low_end,# width of CI
+         above_ci = 100 - high_end) %>% # width between high_end of CI and 100
+  pivot_longer(!c('metric','dummy'), names_to = 'point', values_to = 'value')
+}
+```
+
+```{r data loading and formatting}
+
+# read in performance metrics file
+perf_metrics_data <- read.csv(perf_metrics_filename,
+                              colClasses = c('character', 'character', 'character', 'integer', 'integer')
+                              )
+
+# format funding amounts
+funding_a <- paste0('$',funding_a,'m')
+funding_b <- paste0('$',funding_b,'m')
+funding_c <- paste0('$',funding_c,'m')
+
+```
+
+```{r compute CI measurements}
+
+# compute width and position of CIs for trigger a
+df_a_var <- computeCIwidths(metric_name = 'var', trigger_id = 'a') # valid activation rate 
```

Comment:
I think this section can be simplified with `map2_dfr`, something like: 

```r
df_all = purrr::map2_dfr(
  rep(c("var", "det", "far", "mis", "atv"), times=3),
  rep(c("a", "b", "c"), each=5),
  ~ computeCIwidths(metric_name = .x, trigger_id = .y)
)
```

I think you would need to add the trigger ID to the data frame produced with `computeCIwidths`

Reply:
Yes indeed. I didn't refactor the code and left it as explicit and easy to read as possible for non-R users. I also wanted the documentation to  be clear whereas the map2_dfr is more opaque.

Reply:
Do you expect a lot of non-R users to be using this? I assume it will mostly be us (using the source code at least), so I would advocate for making our own lives more straightforward, especially if we're going to be making several copies of this document. The quick flexibility of `map` would be nice in that case.